src/bun.js/api/bun/dns_resolver.zig

```diff
@@ -1263,28 +1310,47 @@ pub const InternalDNS = struct {
         fn get(
             this: *This,
             key: Request.Key,
+            timestamp_to_store: *u32,
         ) ?*Request {
-            for (this.cache[0..this.len]) |entry| {
+            const len = this.len;
+            for (this.cache[0..len], 0..) |entry, i| {
                 if (entry.key.hash == key.hash and entry.key.port == key.port and entry.valid) {
+                    if (entry.isExpired(timestamp_to_store)) {
+                        log("get: expired entry", .{});
+                        this.deleteEntryAt(len, i);
```

Comment:
This places the last cache entry at `i`, but then increments `i` so I think the newly swapped value will get skipped and never checked